src/client/common/utils/async.ts

```diff
@@ -237,3 +237,25 @@ export async function flattenIterator<T>(iterator: AsyncIterator<T, void>): Prom
     }
     return results;
 }
+
+export class ChainedExecutions<T> {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    private pendingCellExecution: Promise<any> = Promise.resolve();
```

Comment:
Would change the name here, as this class is now general so pendingCellExecution is misleading. Might be something else pending.

Reply:
Also is this a Promise<T> here?

Reply:
NVM, I guess it can't be since it's empy Promse.resolve() at the start.

Reply:
Lold yes